### ISSUE TITLE
You cannot add torrents through a file, only through a magnet

### ISSUE THREAD

**Issue by ori2015** [Michael-128/qBitControl#26]
cannot add torrents through a file, only through a magnet

https://github.com/user-attachments/assets/c40866f6-3862-4188-88fd-8f3e92acbbb1

**Comment by Michael-128** [comment]
Thanks for reporting the bug. Could you tell me what iOS version are you using?

**Comment by ori2015** [comment]
17.6.1 and 18 try both of the version

**Comment by Michael-128** [comment]
Does this also happen for other torrent files or just this specific one?

**Comment by ori2015** [comment]
I checked on all the sites that I have

**Comment by ori2015** [comment]
If it matters then I use the new version via http and not https

**Comment by seannymurrs** [comment]
I'm seeing the same issue. Running iOS 18.1 beta 3.

**Comment by Michael-128** [comment]
Do you have the [newest release](https://github.com/Michael-128/qBitControl/releases/tag/CI) installed? I wasn't able to reproduce the bug before, but if it still happens on the latest version I will look into it.

**Comment by ori2015** [comment]
> Do you have the [newest release](https://github.com/Michael-128/qBitControl/releases/tag/CI) installed? I wasn't able to reproduce the bug before, but if it still happens on the latest version I will look into it.

I downloaded the latest version and still have the same problem

**Comment by Michael-128** [comment]
Is it possible to open the torrent by clicking on it in the Files app? If you have any additional information please be sure to provide them

**Comment by seannymurrs** [comment]
Yes, I’m still seeing the issue with the latest release. When I try to add a task by file, it will let me select the file, but pressing the “Open” button doesn’t actually do anything in the app. Let me know if there’s any other info I can provide to help diagnose further. I can try adding another instance of qBittorrent later today to see if I see the same problem with both installations.

**Comment by seannymurrs** [comment]
When I share via the files app, it opens the app, but it doesn’t actually add the new torrent. I just tried adding my other installation, and I’m seeing the same issue. I have one installation on a seedbox, and one on a local machine at home. I can confirm that both are experiencing the issue. As I previously mentioned, I’m running iOS 18.1 Beta 3. I’ll keep testing different variables (as I can think of them), and I’ll report back if I notice anything different.

**Comment by Michael-128** [comment]
I would try reinstalling the app - completely delete it and install it again. I would also try to download [1.1](https://github.com/Michael-128/qBitControl/releases/tag/1.1) and see if the same thing happens. It would be very helpful to know if the bug is also present in 1.1.

**Comment by Michael-128** [comment]
It's very likely that there is some issue with iOS 18.1 Beta 3 and not the app itself. I'm using the latest iOS 18.0 Beta myself and I don't have this bug. I will set up a simulator with iOS 18.1 and see if the bug is present to be 100% sure.

**Comment by ori2015** [comment]
iOS 18.0 (22A5350a)
This is my version and I have this bag

**Comment by Michael-128** [comment]
@ori2015 That's exactly the same version that I'm using. Could you also check if the bug happens on version 1.1?

**Comment by 68kdmd** [comment]
It works fine for me. 
I’m  on 1.21 and iOS 18.0 (22A3354)

**Comment by ori2015** [comment]
What iPhone do you have? And you use via http or https? And you via the wan or lan?

> It works fine for me. I’m on 1.21 and iOS 18.0 (22A3354)

**Comment by 68kdmd** [comment]
iPhone 13 Pro, http, lan. 

Having used qBitControl since the very first version, I have experienced the issue on and off, and it’s most likely due to the iOS issue with the file handling in their beta versions.

**Comment by ori2015** [comment]
@Michael-128 do you have any idea what I can do?

**Comment by Michael-128** [comment]
@ori2015 Sorry, I don't have any solutions for now. I'm still working on this issue.

**Comment by ori2015** [comment]
Ok thank you very much

**Comment by 68kdmd** [comment]
The fastest solution for you at this point is to change to an iOS that doesn’t have the issue with handling files. See if you can update to iOS 18.0 (22A3354)

**Comment by ori2015** [comment]
I now on iOS 18.0 (22A3354)

**Comment by 68kdmd** [comment]
What model is it?

in the mean time, you could bypass having to add torrent files using Prowlarr or Jacket. I recommend Prowlarr. Add the trackers there and configure your qbittorrent client. Search the torrent in Prowlarr and initiate download.

**Comment by 68kdmd** [comment]
Also, see if you can open a file in other apps like Kodex.

**Comment by ori2015** [comment]
This in kodex is work

**Comment by ori2015** [comment]
Model of what you mean?

**Comment by 68kdmd** [comment]
The same question you asked me before. The model of your iOS device. 

Also, this is going to sound like a generic tech support :D but have you tried restarting your device? It often fixes issues on my File app (connecting to server, accessing directories, and etc).

**Comment by 68kdmd** [comment]
By the  way, what platform are you running your qbittorrent on? And what version? I’m running 4.6.6 on docker. (lscr.io/linuxserver/qbittorrent:latest)

**Comment by ori2015** [comment]
I think it happens because the app is sideload I read that other people have problem loading files to sideload apps is there another way to install the app? iOS 18 non-EU

**Comment by 68kdmd** [comment]
Mine is sideloaded and works perfectly. It’s most likely the iOS issue.

**Comment by ori2015** [comment]
I installed the app through alstore and now it finally works

**Comment by 68kdmd** [comment]
That’s good. What platform is your qbittorrent on? And version?

**Comment by ori2015** [comment]
Windows 4.6.6

**Comment by Tammuz** [comment]
I have the exact same issue. Can't load any file

**Comment by kambala-decapitator** [comment]
adding file works fine for me
